coderd/coderdtest/coderdtest.go

```diff
@@ -92,6 +93,11 @@ func NewWithAPI(t *testing.T, options *Options) (*codersdk.Client, *coderd.API)
 		options.AutobuildTicker = ticker
 		t.Cleanup(func() { close(ticker) })
 	}
+	if options.AutobuildStatsChannel != nil {
+		t.Cleanup(func() {
+			close(options.AutobuildStatsChannel)
```

Comment:
Wondering if this could result in a send on nil channel at some point? Or is the `Run()` function guaranteed to have exited/not tick further at this point?

Reply:
In a unit test, the ticks are sent manually. So, the only way a send could occur on a nil channel is if you did something like

```
t.Cleanup(func() { 
  tickCh <- time.Now()
})
```

Additionally, calls to `t.Cleanup` are executed in LIFO order, so I'd suspect that you might run into `goleak` complaining instead.